foundry.toml

```diff
@@ -4,42 +4,85 @@ out = "out"
 libs = ["lib"]
 optimizer = true
 optimizer_runs = 999999
-via_ir = true
-solc = "0.8.23"
+via_ir = false
+solc = "0.8.26"
 verbosity = 2
 ffi = true
+fs_permissions = [
+  { access = "read-write", path = ".forge-snapshots"},
+  { access = "read", path = "script/" }
+]
 
 remappings = [
   "forge-std=lib/forge-std/src",
   "@openzeppelin/contracts=lib/openzeppelin-contracts/contracts",
   "@openzeppelin/contracts-upgradeable=lib/openzeppelin-contracts-upgradeable/contracts"
 ]
 
-[profile.intense.fuzz]
+[profile.default.fuzz]
+runs = 1000
+
+[profile.pr.fuzz]
 runs = 10000
-max_test_rejects = 999999
+
+[profile.ci.fuzz]
+runs = 100000
+
+[profile.debug]
+via_ir = false
+optimizer_runs = 200
+fuzz.runs = 100
 
 [fmt]
-line_length = 160
+line_length = 120
+tab_width = 4
+quote_style = "single"
+sort_imports = true
 number_underscore = "thousands"
 
 [rpc_endpoints]
 anvil = "http://127.0.0.1:8545"
 mainnet = "https://mainnet.infura.io/v3/${INFURA_KEY}"
-goerli = "https://goerli.infura.io/v3/${INFURA_KEY}"
 sepolia = "https://sepolia.infura.io/v3/${INFURA_KEY}"
+linea = "https://linea-mainnet.infura.io/v3/${INFURA_KEY}"
+linea_sepolia = "https://linea-sepolia.infura.io/v3/${INFURA_KEY}"
 polygon_pos = "https://polygon-mainnet.infura.io/v3/${INFURA_KEY}"
-mumbai = "https://polygon-mumbai.infura.io/v3/${INFURA_KEY}"
+polygon_amoy = "https://polygon-amoy.infura.io/v3/${INFURA_KEY}"
+blast = "https://blast-mainnet.infura.io/v3/${INFURA_KEY}"
+blast_sepolia = "https://blast-sepolia.infura.io/v3/${INFURA_KEY}"
+optimism = "https://optimism-mainnet.infura.io/v3/${INFURA_KEY}"
+optimism_sepolia = "https://optimism-sepolia.infura.io/v3/${INFURA_KEY}"
+arbitrum = "https://arbitrum-mainnet.infura.io/v3/${INFURA_KEY}"
+arbitrum_sepolia = "https://arbitrum-sepolia.infura.io/v3/${INFURA_KEY}"
+celo = "https://celo-mainnet.infura.io/v3/${INFURA_KEY}"
+celo_alfajores = "https://celo-alfajores.infura.io/v3/${INFURA_KEY}"
+zksync = "https://zksync-mainnet.infura.io/v3/${INFURA_KEY}"
```

Comment:
infura doesnt support zksync.. just a callout cause Im just not sure INFURA_KEY will work for every chain

Reply:
seems to be supported now https://docs.infura.io/api/networks/zksync